### DISCUSSION TITLE
Matebook-X-Pro-2018-QOL-Discussion

### DISCUSSION THREAD

**Discussion by tlefko** [tlefko/Matebook-X-Pro-2018-Monterey#12] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by logetec** [comment]
Hey, this may be a stupid question because of the differences of hardware, but would this setup work for the 2020 i7 (Chinese version)? I believe I've used it before but now it gets to the OC picker and reboots after about 10 seconds after choosing MacOS.

**Comment by tlefko** [comment]
@logetec it may work, but I would probably need to edit a couple things as the config is meant for a different GPU / CPU. Would you like me to try to make an EFI and upload?

**Reply by logetec** [reply]
Yeah that'd be great if you can do that! Do you need me to send any info to build the config? Thank you for the quick reply.

**Comment by tlefko** [comment]
Yeah, do you want to list our specs?

—tlefko

**Reply by logetec** [reply]
Yep no problem. I've attached my DxDiag information below. Thanks!
[DxDiag.txt](https://github.com/tlefko/Matebook-X-Pro-2018-Ventura/files/10035174/DxDiag.txt)

**Reply by logetec** [reply]
I have been able to create a "working" config for MBXP 2020 i7 (opencore 0.8.4, anything higher seems to break it?) from another MBXP user's EFI. Everything except bluetooth works and some other small bugs i.e. slow boot times (around 5-10 mins), USB not showing up in finder/location after login. Touchscreen works, function keys work, battery icon shows up. If you would like me to send you the config I can attach it if it would help you to create a "fully functional" config.

**Reply by tlefko** [reply]
@logetec want to send me a copy of the EFI? My email is lefkotyler@gmail.com. Strange that it is breaking above OC 0.8.4, but those bugs sound fixable.

**Reply by logetec** [reply]
Sorry for the late reply, I did send you the email in the end. To fix the slow booting, I set SetApfsTrimTimeout to 0. There is a small inconvenience with the trackpad (for example, clicking the left trackpad button without holding causes the tap to be seen as a double click - I think we can just change the sensitivity in the Voodoo kext right?). I have no clue about the finder/location bug but it seems like the port mapping may be off, because when I use my thunderbolt port for my type-c docking station it doesn't detect any things plugged into that too. USB is detected during the OC picker, but not detected after booting into MacOS.

**Comment by Paxolo** [comment]
The latest release seems to be really stable, you did a very good job! I would daily drive it if it wasn't for a couple issues with audio, bluetooth and AirDrop.
Overall this is hands down the best hackintosh experience I've ever tried, with the os being snappy and thermals that are finally under control on my machine.
Please keep it up! 🙏🏻

**Reply by tlefko** [reply]
@Paxolo thank you for the kind words! I am definitely trying to work on those issues for Ventura, as a lot of that stuff worked fine under monterey. However, airdrop looks like it might be out of the question. Looking to fix bluetooth though and audio! may I ask about the issues you’re having.

**Reply by tlefko** [reply]
also, did you read about layout-id 76? may answer some audio questions.

**Reply by Paxolo** [reply]
I am still under Monterey as of right now, as I haven't had time to update yet.
The main issue I am facing is bluetooth: sometimes it seems to just crash my system.
The worst part is that this happens randomly, and when I go into System Preferences > Bluetooth, it takes a minute to load, just for it to throw an error and close.
Also, once I finally managed to get my AirPods to pair up with my laptop, I realized how poor the quality of the microphone was.
So then I tested some headphones using the audio jack and the quality was even worse.
Maybe, bluetooth problem aside, the bad sound quality of the AirPods microphone has something to do with the audio jack(?)

Last thing is that I did NOT try to use layout-id 76 because I honestly do not know the procedure to do it (what property should it be added to?). However, I ran the script that I thought was supposed to fix audio-related issues.

Thank you SO much for your time :)

**Reply by tlefko** [reply]
@Paxolo okay, so using opencore configurator you can open up the config.plist within the EFI Folder. This will allow you to navigate to this page 
<img width="1312" alt="Screenshot 2022-11-22 at 9 26 56 PM" src="https://user-images.githubusercontent.com/42879340/203458348-1a87365d-7fd0-489d-bf08-ee860fe0eb6a.png">

just change that value to 76. This should fix the audio quality issues (i am assuming you did not setup eqmac) so give this a shot. This is overall the best except it basically breaks the audio jack. As for bluetooth, would you mind sharing which release you're running? If you're using Monterey, you should use the latest release titled that as it should fix the bluetooth. It is strange your audio jack sounded so bad, perhaps the fix did not run correctly. I would run it again if you decide not to change the layout-id and send me the output. Once I know what release you're running I can definitely try to help with the bt issues.

-- No problem, I appreciate the support!

---tlefko

**Comment by catfrog** [comment]
Being a bit confused about the EFI you are talking. 
Does that mean it‘s the default EFI published here?

**Reply by tlefko** [reply]
@catfrog this is on all EFI's. They can all be changed for that layout-id. However, the EFI in question specifically is made for the I7 2020 model

**Comment by catfrog** [comment]
@tlefko unable to boot with the latest EFI for Ventura.
There are a couple of things that I could imagine preventing it.
First of all the the second Grafik card, the resolution of 3 K.
More ore less confirmed by the fact that I didn‘t get the result off the bootarg -v.
Or the case the there are bios changes made which I didn‘t have?
Will check the current boorargs in use, to see what prevents booting.
Is your latest one for the 3K model?
There is one part reserving memory within the config with a given address, does this meet the needs?
Think at least it should help to get a starting point for further investigations.
By now I am guessing in the dark.

**Reply by tlefko** [reply]
@catfrog hmm yes that is odd. this is built for the. 3K display and to be honest isn’t that different from the prior release. When the verbose text stops flowing, if possible try to take a photo and send it here• Likely, I can look over the output to find the bug. There aren’t any BIOS changes that need to be made apart from what was already detailed in the readme. Which part of the config addresses reserved memory? I am slightly confused. once I see the output of verbose I can rule out the Graphic card pretty quickly 

-tlefko

**Comment by mromantan** [comment]
Hi guys, 

Regarding build instructions, it says to download the Catalina Installer, so even if I install correctly, how do I go to Monterey for example?
Many thanks.

**Comment by yonathanarya** [comment]
Hi @tlefko, I want to share my work about enabling the thunderbolt bus (detected in system information) without patching the TBT controller firmware. Currently, I'm using Monterey, as Ventura breaks my wifi scanning. Here is the screenshot.
<img width="1500" alt="Screen Shot 2023-03-29 at 21 48 25" src="https://user-images.githubusercontent.com/18443615/228579140-174def74-c742-47a5-bd96-b234a8547137.png">
Steps to use this:
Remove all ACPI SSDT about TB, or just simply disable it. (**remove all old SSDT-TB*.aml including SSDT-TB3.aml and SSDT-TB3-RP0. Also, disable the SSDT-DTB3.aml as it disables the TB3 controller**)
Copy all content from my ACPI folder below to your OC/ACPI folder and enable everything on config.plist
Copy the TbtForcePower.efi to OC/DRIVERS and enable it on config.plist.
Sorry, forgot to mention that It also need DSDT ACPI Patch on config.plist
Find -> 5F453233 -> Replace -> 58453233
Find -> 5F494E49 -> Replace -> 58494E49

And it's done now. The issues that I'm facing is when the thunderbolt bus is enabled and recognized in System Information, my thunderbolt eGPU doesn't work (macOS freeze and kernel panic occurred). If you don't use eGPU, it's fine to use the SSDT-TB-Matebook.aml but if you use it, just disable it at the cost of losing thunderbolt bus support for other thing like file transfer between mac device through thunderbolt, internet connection sharing, etc. The other issue is, after sleep the thunderbolt NHI controller doesn't recognized anymore but the USB 3.1 bus is still working.

All of this work hotplug.

PS: If you don't plan to use eGPU, just use the SSDT-TB-Matebook.aml and ignore the other SSDT, but you still need the efi driver anyway to force power on TBT controller. Feel free to add this on your repo if you think it's worth it.

[ENABLE_TB_DRIVER_MBXP2018.zip](https://github.com/tlefko/Huawei-Matebook-X-Pro-Ventura/files/11102238/ENABLE_TB_DRIVER_MBXP2018.zip)

**Reply by tlefko** [reply]
> Hi @tlefko, I want to share my work about enabling the thunderbolt bus (detected in system information) without patching the TBT controller firmware. Currently, I'm using Monterey, as Ventura breaks my wifi scanning. Here is the screenshot. <img alt="Screen Shot 2023-03-29 at 21 48 25" width="1500" src="https://user-images.githubusercontent.com/18443615/228579140-174def74-c742-47a5-bd96-b234a8547137.png"> Steps to use this: Remove all ACPI SSDT about TB, or just simply disable it. (__remove all old SSDT-TB_.aml including SSDT-TB3.aml and SSDT-TB3-RP0. Also, disable the SSDT-DTB3.aml as it disables the TB3 controller_*) Copy all content from my ACPI folder below to your OC/ACPI folder and enable everything on config.plist Copy the TbtForcePower.efi to OC/DRIVERS and enable it on config.plist. And it's done now. The issues that I'm facing is when the thunderbolt bus is enabled and recognized in System Information, my thunderbolt eGPU doesn't work (macOS freeze and kernel panic occurred). If you don't use eGPU, it's fine to use the SSDT-TB-Matebook.aml but if you use it, just disable it at the cost of losing thunderbolt bus support for other thing like file transfer between mac device through thunderbolt, internet connection sharing, etc. The other issue is, after sleep the thunderbolt NHI controller doesn't recognized anymore but the USB 3.1 bus is still working.
> 
> All of this work hotplug.
> 
> PS: If you don't plan to use eGPU, just use the SSDT-TB-Matebook.aml and ignore the other SSDT, but you still need the efi driver anyway to force power on TBT controller. Feel free to add this on your repo if you think it's worth it.
> 
> [ENABLE_TB_DRIVER_MBXP2018.zip](https://github.com/tlefko/Huawei-Matebook-X-Pro-Ventura/files/11102238/ENABLE_TB_DRIVER_MBXP2018.zip)

@profzei

**Comment by tlefko** [comment]
@yonathanarya thank you for publishing this, I am going to test out this out and see if I can make some improvements. if we can get the controller to persist after sleep, I think that would be a great step. Also, do you have a copy of any of the kernel panic logs for the EGPU connection? -- The wifi scanning failiure I believe is related to the wifi kext, not TBT.  I am not entirely sure why that's causing issues, does it also freeze if u do it after sleep?

Will attempt to force the controller as always awake perhaps.

**Reply by yonathanarya** [reply]
I know the wifi is caused by Airportitlwm, cause it has been reported by many users in ventura. Anyway, here is my kernel panic log:

`{"timestamp":"2023-03-29 20:52:51.00 +0700","bug_type":"210","os_version":"macOS 12.6.3 (21G419)","incident_id":"3A2DD76F-F1CC-4E6D-87CB-A218718BC98F"}
{"macOSProcessedStackshotData":"bm8gb24gZGlzayBwYW5pYyBzdGFja3Nob3QgZm91bmQgaW4gY29yZWZpbGU=","macOSPanicString":"panic(cpu 0 caller 0xffffff8010bcdd43): Kernel trap at 0xffffff8010be363a, type 13=general protection, registers:\nCR0: 0x000000008001003b, CR2: 0xffffffffe0358000, CR3: 0x0000000028dd0000, CR4: 0x00000000003626e0\nRAX: 0x000000007e008003, RBX: 0xffffff8011899140, RCX: 0x00000000000000e2, RDX: 0x0000000000000000\nRSP: 0xffffffadaeb63ef0, RBP: 0xffffffadaeb63f20, RSI: 0x000000007e008000, RDI: 0x000000007e008003\nR8:  0x0000000000000000, R9:  0x0000000000000008, R10: 0x0000000000000000, R11: 0xffffff8011a63980\nR12: 0xffffff8011899140, R13: 0x00000000012e483c, R14: 0x0000000274f53f2a, R15: 0x0000000000000000\nRFL: 0x0000000000010046, RIP: 0xffffff8010be363a, CS:  0x0000000000000008, SS:  0x0000000000000010\nFault CR2: 0xffffffffe0358000, Error code: 0x0000000000000000, Fault CPU: 0x0, PL: 1, VF: 0\n\nPanicked task 0xffffff9512ff1670: 191 threads: pid 0: kernel_task\nBacktrace (CPU 0), panicked thread: 0xffffff86ad96e540, Frame : Return Address\n0xffffff801090c170 : 0xffffff8010a7bebd \n0xffffff801090c1c0 : 0xffffff8010bde5e6 \n0xffffff801090c200 : 0xffffff8010bcd953 \n0xffffff801090c250 : 0xffffff8010a1ba70 \n0xffffff801090c270 : 0xffffff8010a7c28d \n0xffffff801090c390 : 0xffffff8010a7ba46 \n0xffffff801090c3f0 : 0xffffff80113148b3 \n0xffffff801090c4e0 : 0xffffff8010bcdd43 \n0xffffff801090c660 : 0xffffff8010bcda26 \n0xffffff801090c6b0 : 0xffffff8010a1ba70 \n0xffffff801090c6d0 : 0xffffff8010be363a \n0xffffffadaeb63f20 : 0xffffff8010bcff56 \n0xffffffadaeb63f40 : 0xffffff8010aaba3a \n0xffffffadaeb63f80 : 0xffffff8010aabc39 \n0xffffffadaeb63fa0 : 0xffffff8010a1b19e \n\nProcess name corresponding to current thread (0xffffff86ad96e540): kernel_task\nBoot args: alcid=76 agdpmod=pikera watchdog=0 -raddvi\n\nMac OS version:\n21G419\n\nKernel version:\nDarwin Kernel Version 21.6.0: Mon Dec 19 20:44:01 PST 2022; root:xnu-8020.240.18~2\/RELEASE_X86_64\nKernel UUID: 33C30416-3302-3FFF-9686-A557DED28C5B\nKernelCache slide: 0x0000000010800000\nKernelCache base:  0xffffff8010a00000\nKernel slide:      0x0000000010810000\nKernel text base:  0xffffff8010a10000\n__HIB  text base: 0xffffff8010900000\nSystem model name: MacBookPro16,3 (Mac-E7203C0F68AA0004)\nSystem shutdown begun: NO\nPanic diags file available: YES (0x0)\nHibernation exit count: 0\n\nSystem uptime in nanoseconds: 10552264236\nLast Sleep:           absolute           base_tsc          base_nano\n  Uptime  : 0x0000000274f6cd51\n  Sleep   : 0x0000000000000000 0x0000000000000000 0x0000000000000000\n  Wake    : 0x0000000000000000 0x00000004aa2709ae 0x0000000000000000\nCompressor Info: 0% of compressed pages limit (OK) and 0% of segments limit (OK) with 0 swapfiles and OK swap space\nZone info:\n  Zone map: 0xffffff80458a8000 - 0xffffffa0458a8000\n  . PGZ   : 0xffffff80458a8000 - 0xffffff80478a9000\n  . VM    : 0xffffff80478a9000 - 0xffffff85140a8000\n  . RO    : 0xffffff85140a8000 - 0xffffff86ad8a8000\n  . GEN0  : 0xffffff86ad8a8000 - 0xffffff8b7a0a8000\n  . GEN1  : 0xffffff8b7a0a8000 - 0xffffff90468a8000\n  . GEN2  : 0xffffff90468a8000 - 0xffffff95130a8000\n  . GEN3  : 0xffffff95130a8000 - 0xffffff99df8a8000\n  . DATA  : 0xffffff99df8a8000 - 0xffffffa0458a8000\n  Metadata: 0xffffffadc0da9000 - 0xffffffade0da9000\n  Bitmaps : 0xffffffade0da9000 - 0xffffffade6da9000\n\n\n\n"}
`

**Comment by tlefko** [comment]
@yonathanarya it is a USB issue, I think it has to do with the host controller. It is hard to tell this is a super generic error, and I don't have an EGPU to test, but I would try something really simple first. Enable XHCI port limit within the config.plist. I would also enable some of the previous thunderbolt patches and give it a shot and let me know where we end up. Next idea will be to try to change the actual thunderbolt software in macOS because ik there's a way to actually bypass the native software-controller. Also make sure DGPT patch is on

**Reply by yonathanarya** [reply]
i don't really know if it just a USB issue or not. Because when I try to use other TB Device like NVME Enclosure it just works. 
<img width="1500" alt="Screen Shot 2023-03-29 at 22 36 13" src="https://user-images.githubusercontent.com/18443615/228591727-b091d60f-2699-446d-8408-ea1c7326ef4c.png">

**Reply by tlefko** [reply]
> i don't really know if it just a USB issue or not. Because when I try to use other TB Device like NVME Enclosure it just works.
> <img alt="Screen Shot 2023-03-29 at 22 36 13" width="1500" src="https://user-images.githubusercontent.com/18443615/228591727-b091d60f-2699-446d-8408-ea1c7326ef4c.png">

Will look for TB devices to test and troubleshoot

**Comment by yonathanarya** [comment]
I have another suggestion anyway. If you could, try to remove DSDT.aml and just replace it with SSDT patch. Because in my experience, I can't boot to Ventura with your DSDT or even just update OC and stay at Monterey, because sometimes DSDT only works on a certain device only and not working on another device. I tried removing DSDT and changing it with some SSDT patch in it can boot to Ventura without any hassle. Thanks for your work anyway.

**Comment by tlefko** [comment]
Hey, this is actually really surprising to me. I don't know what to replace
it with as it essentially is a ridiculous amount of ssdt patches in one
file. I need the file in order to boot, as without it i kernel panic. What
did you use for ventura as opposed to it?

**Reply by yonathanarya** [reply]
I replace the dsdt with ssdt from olarila kabylake desktop. It can boot but without the brightness control

**Comment by tlefko** [comment]
Ok, i will add that to the file and publish.
